pkg/config/validater.go

```diff
@@ -0,0 +1,220 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/yaml.v2"
+
+	"istio.io/mixer/pkg/aspect"
+	"istio.io/mixer/pkg/expr"
+)
+
+type (
+	// ValidaterFinder is used to find specific underlying validaters
+	// Manager registry and adapter registry should implement this interface
+	// so ConfigValidaters can be uniformly accessed
+	ValidaterFinder interface {
+		Find(name string) (aspect.ConfigValidater, bool)
+	}
+)
+
+// NewValidater returns a validater given component validaters
+func NewValidater(managerFinder ValidaterFinder, adapterFinder ValidaterFinder, strict bool, exprValidater expr.Validater) *Validater {
+	return &Validater{
+		managerFinder: managerFinder,
+		adapterFinder: adapterFinder,
+		strict:        strict,
+		exprValidater: exprValidater,
+		validated:     &Validated{},
+	}
+}
+
+type (
+	// Validater is the Configuration validator
+	Validater struct {
+		managerFinder ValidaterFinder
+		adapterFinder ValidaterFinder
+		strict        bool
+		exprValidater expr.Validater
+		validated     *Validated
+	}
+
+	// Validated store validated configuration
+	// It has been validated as internally consistent and correct
+	Validated struct {
+		// Names are given to specific adapter configurations: unique
+		adapterByName map[string]*Adapter
+		// This is more often used to match adapters
+		adapterByKind map[string][]*Adapter
+		serviceConfig *ServiceConfig
+		numAspects    int
+	}
+)
+
+// ValidateAdapterConfig consumes a yml config string with adapter config
+// It is validated in presence of
+func (p *Validater) validateGlobalConfig(cfg string) (ce *aspect.ConfigErrors) {
+	m := &GlobalConfig{}
+	err := yaml.Unmarshal([]byte(cfg), m)
+	if err != nil {
+		ce = ce.Append("AdapterConfig", err)
+		return
+	}
+	p.validated.adapterByKind = make(map[string][]*Adapter)
+	p.validated.adapterByName = make(map[string]*Adapter)
+	var acfg proto.Message
+	var aArr []*Adapter
+	var found bool
+	for _, aa := range m.GetAdapters() {
+		if acfg, err = ConvertParams(p.adapterFinder, aa.Impl, aa.Params, p.strict); err != nil {
+			ce = ce.Append("Adapter: "+aa.Impl, err)
+			continue
+		}
+		aa.Params = acfg
+
+		if aArr, found = p.validated.adapterByKind[aa.GetKind()]; !found {
+			aArr = []*Adapter{}
+		}
+		aArr = append(aArr, aa)
+		p.validated.adapterByKind[aa.GetKind()] = aArr
+		if aa.GetName() != "" {
+			p.validated.adapterByName[aa.GetName()] = aa
+		}
+
+	}
+	return
+}
+
+// ValidateSelector ensures that the selector is valid per expression language
+func (p *Validater) validateSelector(selector string) (err error) {
+	if len(selector) == 0 {
+		return fmt.Errorf("empty selector not allowed")
+	}
+	return p.exprValidater.Validate(selector)
+}
+
+// ValidateRules  processes Aspect Rules
```

Comment:
extra space here.

Can we also add a bit more about what a valid rule is maybe?